app/IATI/Services/ImportActivity/ImportXmlService.php

```diff
@@ -190,8 +193,8 @@ public function create($activities): bool
                 $this->activityRepository->importXmlActivities($oldActivity->id, $activityData);
                 $this->transactionRepository->deleteTransaction($oldActivity->id);
                 $this->resultRepository->deleteResult($oldActivity->id);
-                $this->saveTransactions(Arr::get($activityData, 'transactions'), $oldActivity->id);
-                $this->saveResults(Arr::get($activityData, 'result'), $oldActivity->id);
+                $this->saveTransactions(Arr::get($activityData, 'transactions'), $oldActivity->id, Arr::get($activityData, 'default_field_values.0', []));
```

Comment:
use variable to represent default_field_values instead of accessing it multiple times